pdaltools/add_points_in_pointcloud.py

```diff
@@ -35,37 +37,80 @@ def parse_args(argv=None):
         default=1000,
         help="width of tiles in meters",
     )
+    parser.add_argument(
+        "--spacing",
+        type=float,
+        default=0,
+        help="spacing between generated points in meters",
+    )
+    parser.add_argument(
+        "--altitude_column",
+        "-z",
+        type=str,
+        required=True,
+        default="RecupZ",
+        help="altitude column name from input geometry",
+    )
 
     return parser.parse_args(argv)
 
 
-def clip_3d_points_to_tile(input_points: str, input_las: str, crs: str, tile_width: int) -> gpd.GeoDataFrame:
+def clip_3d_lines_to_tile(
+    input_lines: gpd.GeoDataFrame, input_las: str, crs: str, tile_width: int
+) -> gpd.GeoDataFrame:
     """
-    Add points from a GeoJSON file in the LIDAR's tile.
+    Clip lines from a GeoDataFrame to the LIDAR's tile.
 
     Args:
-        input_points (str): Path to the input GeoJSON file with 3D points.
+        input_lines (gpd.GeoDataFrame): GeoDataFrame with lines.
         input_las (str): Path to the LIDAR `.las/.laz` file.
         crs (str): CRS of the data.
         tile_width (int): Width of the tile in meters (default: 1000).
 
-    Return:
-        gpd.GeoDataFrame: Points 2d with "Z" value
+    Returns:
+        gpd.GeoDataFrame: Lines that intersect with the tile.
     """
     # Compute the bounding box of the LIDAR tile
     tile_bbox = get_tile_bbox(input_las, tile_width)
 
-    # Read the input GeoJSON with 3D points
-    points_gdf = gpd.read_file(input_points)
+    if crs:
+        input_lines = input_lines.to_crs(crs)
+
+    # Create a polygon from the bounding box
+    bbox_polygon = box(*tile_bbox)
+
+    # Clip the lines to the bounding box
+    clipped_lines = input_lines[input_lines.intersects(bbox_polygon)].copy()
+
+    return clipped_lines
+
+
+def clip_3d_points_to_tile(
+    input_points: gpd.GeoDataFrame, input_las: str, crs: str, tile_width: int
+) -> gpd.GeoDataFrame:
+    """
+    Add points from a GeoDataFrame in the LIDAR's tile.
```

Comment:
la docstring correspond pas exactement.
Proposition pour qqc de plus proche de la réalité : 
```suggestion
    Select points from a GeoDataFrame that intersec the LIDAR tile
```